js/weechat.js

```diff
@@ -628,17 +628,51 @@
     };
 
     /**
-     * Formats an init command.
+     * Formats a handshake command.
+     *
+     * @param params Parameters:
+     *            password: list of supported hash algorithems, colon separated (optional)
+     *            compression: compression ('off' or 'zlib') (optional)
+     * @return Formatted handshake command string
+     */
+    //https://weechat.org/files/doc/stable/weechat_relay_protocol.en.html#command_handshake
+    WeeChatProtocol.formatHandshake = function(params) {
+        var defaultParams = {
+            password_hash_algo: 'pbkdf2+sha512',
+            compression: 'zlib'
+        };
+        var keys = [];
+        var parts = [];
+
+        params = WeeChatProtocol._mergeParams(defaultParams, params);
+
+        if (params.password !== null) {
```

Comment:
why do these depend on the password not being null?

Reply:
Two seperate mistakes, "password" used to be what "password_hash_algo" is now, forgot to change that. And the compression should always have been compression.

Updated, fixed, tested.